ipinfo/cmd_asn.go

```diff
@@ -67,6 +68,9 @@ func cmdASN(asn string) error {
 
 	data, err := ii.GetASNDetails(asn)
 	if err != nil {
+		if strings.Contains(err.Error(), "401 { }") {
```

Comment:
It'd be nice if we could get at the underlying error type and refer to it directly. Parsing the error string like this definitely feels like a hack. Unless the underlying error type is storing any important info as a string only and no other form.

Reply:
The underlying error is just a string without any additional information that we can use. that's why this was the only possible solution I could figure out. 
```
string
err: GET https://ipinfo.io/AS63987: 401 { }
```

Reply:
See the IPinfo Go SDK. Follow the logic carefully from the `GetASNDetails` function call and you will find the actual error type. It is a real error type and contains even the full Response object in it.